packages/components/src/Menu/Menu.tsx

```diff
@@ -111,34 +118,45 @@ export function Menu({
   useRefocusOnActivator(visible);
   const menuRef = useFocusTrap<HTMLDivElement>(visible);
 
-  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
-  const {
-    styles: popperStyles,
-    attributes,
-    state,
-  } = usePopper(popperRef.current, popperElement, {
+  const { refs, floatingStyles, context } = useFloating({
+    open: visible,
+    onOpenChange: setVisible,
     placement: "bottom-start",
     strategy: "fixed",
-    modifiers: [
-      {
-        name: "flip",
-        options: {
-          flipVariations: true,
+    middleware: [
+      offset(MENU_OFFSET),
+      flip({ flipAlignment: false }),
+      size({
+        apply({ availableHeight, elements }) {
+          // The inner element is the scrollable menu that requires the max height
+          const menuElement = elements.floating.querySelector(
+            '[role="menu"]',
+          ) as HTMLElement;
+
+          if (menuElement) {
+            const viewportHeight = window.innerHeight;
+            const maxHeightVh = (viewportHeight * MENU_MAX_HEIGHT_VH) / 100;
+
+            // Use the smaller of 72vh or available space
+            const maxHeight = Math.min(maxHeightVh, availableHeight);
+
+            Object.assign(menuElement.style, {
+              maxHeight: `${maxHeight}px`,
+            });
+          }
         },
-      },
-      {
-        name: "offset",
-        options: {
-          offset: [0, MENU_OFFSET],
-        },
-      },
+      }),
     ],
+    elements: {
+      reference: referenceElement,
+    },
+    whileElementsMounted: autoUpdate,
   });
+
   const positionAttributes =
     width >= SMALL_SCREEN_BREAKPOINT
       ? {
-          ...attributes.popper,
```

Comment:
once again, we're "losing" this but it's just popper stuff

<img width="346" height="186" alt="image" src="https://github.com/user-attachments/assets/848770ae-5e7e-4521-aba3-83635e0f600a" />